integrations/google_genai/src/haystack_integrations/components/generators/google_genai/chat/chat_generator.py

```diff
@@ -0,0 +1,579 @@
+    # and remove the $defs key leaving the rest of the schema
+    tool_schema = remove_key_from_schema(tool_schema, "$defs")
+
+    return tool_schema
+
+
+def _convert_tools_to_google_genai_format(tools: Union[List[Tool], Toolset]) -> List[types.Tool]:
+    """
+    Converts a list of Haystack Tools or a Toolset to Google Gen AI Tool format.
+
+    :param tools: List of Haystack Tool objects or a Toolset.
+    :returns: List of Google Gen AI Tool objects.
+    """
+    # Convert Toolset to list if needed
+    if isinstance(tools, Toolset):
+        tools = list(tools)
+
+    function_declarations = []
+    for tool in tools:
+        parameters = _sanitize_tool_schema(tool.parameters)
+        function_declarations.append({"name": tool.name, "description": tool.description, "parameters": parameters})
+
+    # Return a single Tool object with all function declarations as in the Google GenAI docs
+    # we could also return multiple Tool objects, doesn't seem to make a difference
+    # revisit this decision
+    return [types.Tool(function_declarations=function_declarations)]  # type: ignore[arg-type]
+
+
+def _convert_google_genai_response_to_chatmessage(response, model: str) -> ChatMessage:
+    """
+    Converts a Google Gen AI response to a Haystack ChatMessage.
+
+    :param response: The response from Google Gen AI.
+    :param model: The model name.
+    :returns: A Haystack ChatMessage.
+    """
+    text_parts = []
+    tool_calls = []
+
+    # Extract text and function calls from response
+    finish_reason = None
+    if response.candidates:
+        candidate = response.candidates[0]
+        finish_reason = getattr(candidate, "finish_reason", None)
+        for part in candidate.content.parts:
+            if part.text is not None:
+                text_parts.append(part.text)
+            if part.function_call is not None:
+                tool_call = ToolCall(
+                    tool_name=part.function_call.name,
+                    arguments=dict(part.function_call.args) if part.function_call.args else {},
+                    id=part.function_call.id,
+                )
+                tool_calls.append(tool_call)
+
+    # Combine text parts
+    text = " ".join(text_parts) if text_parts else ""
+
+    # Create ChatMessage
+    message = ChatMessage.from_assistant(text=text, tool_calls=tool_calls)
+
+    # Add metadata
+    message._meta.update(
+        {
+            "model": model,
+            "finish_reason": str(finish_reason) if finish_reason else None,
+        }
+    )
+
+    return message
+
+
+@component
+class GoogleGenAIChatGenerator:
+    """
+    A component for generating chat completions using Google's Gemini models via the Google Gen AI SDK.
+
+    This component provides an interface to Google's Gemini models through the new google-genai SDK,
+    supporting models like gemini-2.0-flash and other Gemini variants.
+
+    ### Usage example
+
+    ```python
+    from haystack.utils import Secret
```

Comment:
I'd leave out the Secret here and also remove `Secret.from_env_var("GOOGLE_API_KEY")` from the example for simplicity